core/templates/dev/head/editor/feedback_tab/ThreadDataService.js

```diff
@@ -31,18 +31,24 @@ oppia.factory('threadDataService', [
   // objects, each representing threads. The 'messages' key of this object
   // is updated lazily.
   var _data = {
-    threadList: []
+    feedbackThreads: [],
+    suggestionThreads: []
   };
 
   var _fetchThreads = function(successCallback) {
-    // UI test code: http://pastebin.com/DhL4fGnm
     var fPromise = $http.get(_THREAD_LIST_HANDLER_URL);
     var sPromise = $http.get(_SUGGESTION_LIST_HANDLER_URL, {
-      params: {type: 'open'}
+      params: {list_type: 'all', has_suggestion: true}
     });
 
     $q.all([fPromise, sPromise]).then(function(res) {
-      _data.threadList = [].concat(res[0].data.threads, res[1].data.threads);
+      _data.feedbackThreads = res[0].data.threads;
+      _data.suggestionThreads = res[1].data.threads;
+      // Later requests will use only the thread_id, not full_thread_id
```

Comment:
Yep! If we can confine the use of full_thread_id to as deep in the stack as possible, and not have it percolate upwards, that would be best.